src/components/TextLink.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {FC, ForwardedRef, KeyboardEventHandler, MouseEventHandler, ReactElement} from 'react';
+import {GestureResponderEvent, Text as RNText, TextStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import Text from './Text';
+
+type TextLinkProps = {
+    /** Link to open in new tab */
+    href: string;
+
+    /** Text content child */
+    children: ReactElement;
+
+    /** Additional style props */
+    style?: TextStyle;
+
+    /** Overwrites the default link behavior with a custom callback */
+    onPress?: () => void;
+
+    /** Callback that is called when mousedown is triggered */
+    onMouseDown?: MouseEventHandler;
+
+    /** A ref to forward to text */
+    forwardedRef: ForwardedRef<RNText>;
+};
+
+function TextLink({href, children, style, onPress, onMouseDown = (event) => event.preventDefault(), forwardedRef, ...props}: TextLinkProps) {
+    const openLink = () => {
+        if (onPress) {
+            onPress();
+            return;
+        }
+
+        Link.openExternalLink(href);
+    };
+
+    const openLinkOnTap = (event: GestureResponderEvent) => {
+        event.preventDefault();
+
+        openLink();
+    };
+
+    const openLinkOnEnterKey: KeyboardEventHandler = (event) => {
+        event.preventDefault();
```

Comment:
Why `event.preventDefault();` was moved from openLink? I advise against logic changes in TS migrations

Reply:
I hope there's no logic change in here. I couldn't call `openLink(event)` from `openLinkIfEnterKeyPressed` function because the `event` would be of different type than what `openLink` expects (`KeyboardEvent` vs `GestureResponderEvent`). So I removed `event.preventDefault()` logic from `openLink` and now I don't need to pass any arguments to `openLink()` because `event.preventDefault()` is being called from `openLinkOnTap()` and `openLinkOnEnterKey`. LMK if that makes sense. The only logic change here is that I technically changed the order of when `event.preventDefault()` is called in `openLinkOnEnterKey`:
Was:
```
    const openLinkOnEnterKey: KeyboardEventHandler = (event) => {
        event.preventDefault();
        if (event.key !== 'Enter') {
            return;
        }

        openLink();
    };
```
Should be:
```
    const openLinkOnEnterKey: KeyboardEventHandler = (event) => {
        if (event.key !== 'Enter') {
            return;
        }
        event.preventDefault();

        openLink();
    };
```

Reply:
I'll move `event.preventDefault()` below the early return.